e2e/tests/uniswap_trade_test.rs

```diff
@@ -246,6 +246,7 @@ async fn test_with_ganache() {
         uniswap_like_liquidity: vec![uniswap_liquidity],
         orderbook_api: create_orderbook_api(&web3),
     };
+    let network_id = web3.net().version().await.unwrap();
```

Comment:
We should probably start using chain ID instead of network ID everywhere, especially since it is tied to the signatures.

Just a general comment that doesn't have to be addressed in this PR.